modules/infra/control/iface.c

```diff
@@ -96,9 +98,15 @@ struct iface *iface_create(const struct gr_iface *conf, const void *api_info) {
 		ifid = conf->vrf_id;
 		if (reserve_ifid(ifid) < 0)
 			goto fail;
-	} else if (next_ifid(&ifid) < 0)
-		goto fail;
+	} else {
+		if (vrf_incref(conf->vrf_id) < 0)
+			goto fail;
+		vrf_ref = true;
+		iface->vrf_id = conf->vrf_id;
 
+		if (next_ifid(&ifid) < 0)
+			goto fail;
+	}
```

Comment:
This is not correct. You may hit this code branch with `conf->type == GR_IFACE_TYPE_LOOPBACK && conf->vrf_id == 0`. This should be changed to:

```c
	if (conf->type != GR_IFACE_TYPE_LOOPBACK) {
		if (vrf_incref(conf->vrf_id) < 0)
			goto fail;
		vrf_ref = true;
	}

	if (conf->type == GR_IFACE_TYPE_LOOPBACK && conf->vrf_id) {
		ifid = conf->vrf_id;
		if (reserve_ifid(ifid) < 0)
			goto fail;
	} else if (next_ifid(&ifid) < 0)
		goto fail;
```